### ISSUE TITLE
Prometheus Plugin not support chinese characters

### ISSUE THREAD

**Issue by havengit** [apache/cloudstack#8203]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Prometheus Plugin
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### SUMMARY

When using Chinese in the domain name, the Prometheus interface request does not response

##### STEPS TO REPRODUCE

~~~
create a new domian with name "中文"
Then curl http://server:9595/metrics
~~~

##### EXPECTED RESULTS

~~~
Return data
~~~

##### ACTUAL RESULTS

~~~
No response
~~~

**Comment by havengit** [comment]
When the domian is set to Chinese characters, only the exporter is abnormal, and the cloudstack operation is normal.
Only the client request events  are found in the logs,  no exception logs.

log for prometheus:

root@mgr1dom1:~# cat /var/log/cloudstack/management/management-server.log |grep -i prom
2023-11-10 01:18:02,066 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:30:19,049 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:31:06,168 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.226.15.101
2023-11-10 01:31:17,386 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.226.15.101
2023-11-10 01:32:01,297 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:32:08,566 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:32:30,801 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:32:44,211 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33
2023-11-10 01:34:02,871 DEBUG [o.a.c.m.PrometheusExporterServerImpl] (HTTP-Dispatcher:null) (logid:) Prometheus exporter received client request from: 10.223.96.33

![image](https://github.com/apache/cloudstack/assets/20832243/65501086-63d3-4a18-ad40-43ec2d81792e)

**Comment by DaanHoogland** [comment]
@havengit , few more
- does your prometheus import any data from other systems?
- can you create accounts/network/VMs with Chinese (or other language) characters?

**Comment by havengit** [comment]
- I just enable prometheus in cloudstack,did not import any data from other system
- create account and network with chinese characters ,No error occurred when requesting ip:9595/metrics
- In metrics there are only domain and zone, which are user-defined fields, so when these use Chinese characters, the request metrics does not respond.

![image](https://github.com/apache/cloudstack/assets/20832243/53e93456-eab2-4f37-a31c-bbeb14e9827b)

**Comment by DaanHoogland** [comment]
clear, I'll have a test with for instance ë to see if it is specific to a unicode range. This will probably need a fix in the code (as you already figured probably ;).

**Comment by havengit** [comment]
Great, looking forward to the fix!

**Comment by DaanHoogland** [comment]
I can confirm that a domain named "één" or "geïntegreerd" does not work either (about the extend of the difference in character sets between dutch and english;).

**Comment by DaanHoogland** [comment]
if you can please test #8228 @havengit

**Comment by havengit** [comment]
Roger that. I'll test it and get back to you.

**Comment by havengit** [comment]
I have test this pr,  There are some improvements, it can be responsive when using Chinese characters, but it will mess up the code.

EXPECTED RESULTS:
“中文”
ACTUAL RESULTS:
”涓 枃“

![image](https://github.com/apache/cloudstack/assets/20832243/a3d96744-af6f-4ebf-8869-57baa35bdc05)
![image](https://github.com/apache/cloudstack/assets/20832243/793e7056-c950-481f-a9c2-536132decce1)

**Comment by havengit** [comment]
Hi I change this line ,then It works perfect.
httpExchange.getResponseHeaders().set("content-type", "text/plain");
httpExchange.getResponseHeaders().set("content-type", "text/plain; charset=UTF-8");
![image](https://github.com/apache/cloudstack/assets/20832243/d379642c-39b5-49ce-ba11-049ec9ecc0df)

**Comment by DaanHoogland** [comment]
thanks @havengit , after smoke tests pass I will merge it.

**Comment by weizhouapache** [comment]
@DaanHoogland 
would you like to apply the changes in @havengit's comment ?
```
httpExchange.getResponseHeaders().set("content-type", "text/plain; charset=UTF-8");
```

**Comment by DaanHoogland** [comment]
> @DaanHoogland would you like to apply the changes in @havengit's comment ?
> 
> ```
> httpExchange.getResponseHeaders().set("content-type", "text/plain; charset=UTF-8");
> ```

right, i missed that. Strange, it worked in my test. Will apply.

**Comment by DaanHoogland** [comment]
fixed by #8228
